grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcService.java

```diff
@@ -188,15 +191,17 @@ protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req) throws
                             false));
         }
 
-        final String timeoutHeader = req.headers().get(GrpcHeaderNames.GRPC_TIMEOUT);
-        if (timeoutHeader != null) {
-            try {
-                final long timeout = TimeoutHeaderUtil.fromHeaderValue(timeoutHeader);
-                ctx.setRequestTimeout(Duration.ofNanos(timeout));
-            } catch (IllegalArgumentException e) {
-                return HttpResponse.of(
-                        ArmeriaServerCall.statusToTrailers(
-                                ctx, GrpcStatus.fromThrowable(e), new Metadata(), false));
+        if (useClientTimeoutHeader) {
+            final String timeoutHeader = req.headers().get(GrpcHeaderNames.GRPC_TIMEOUT);
+            if (timeoutHeader != null) {
+                try {
+                    final long timeout = TimeoutHeaderUtil.fromHeaderValue(timeoutHeader);
+                    ctx.setRequestTimeout(Duration.ofNanos(timeout));
+                } catch (IllegalArgumentException e) {
+                    return HttpResponse.of(
+                            ArmeriaServerCall.statusToTrailers(
+                                    ctx, GrpcStatus.fromThrowable(e), new Metadata(), false));
```

Comment:
nit: This calls `static HttpResponse of(HttpObject... objs) {...}` instead of `static HttpResponse of(ResponseHeaders headers) {...}`. How about type-casting?
```java
return HttpResponse.of(
        (ResponseHeaders) ArmeriaServerCall.statusToTrailers(
                ctx, GrpcStatus.fromThrowable(e), new Metadata(), false));
```

Reply:
Sure added the cast